device/cuda/src/clusterization/clusterization_algorithm.cu

```diff
@@ -152,7 +152,6 @@ clusterization_algorithm::output_type clusterization_algorithm::operator()(
         spacepoints_buffer);
 
     CUDA_ERROR_CHECK(cudaGetLastError());
-    m_stream.synchronize();
```

Comment:
I think this is a good change. But you should add some explicit mention about this in the documentation of this algorithm.

Namely that the users of the algorithm **must not** dismiss the return value of the algorithm, or delete it before the `form_spacepoints` kernel would've finished running.

The way that vecmem buffers work, it **is** possible to pass around objects between components that are in the process of being filled. We'll just need to do this carefully.